cmd/root.go

```diff
@@ -1131,7 +1134,10 @@ func runSignalWrapper(cmd *Command) (err error) {
 		)
 	}
 
-	go func() { shutdownCh <- p.Serve(ctx, notifyStarted) }()
+	go func() {
+		cmd.logger.Debugf("Starting to listen on shutdown channel.")
```

Comment:
By logging the error, we can see what Serve might be complaining about and then infer if the error is reported through the channel based on subsequent log messages.

``` golang
err := <- p.Serve(ctx, notifyStarted)
cmd.logger.Debugf("proxy server error: %v", err)
shutdownCh <- err
```

Reply:
did `err := p.Serve(ctx, notifyStarted)` instead of `err := <- p.Serve(ctx, notifyStarted)`

Reply:
Yep you're correct.